line-sdk/src/main/java/com/linecorp/linesdk/openchat/ui/OpenChatInfoViewModel.kt

```diff
@@ -0,0 +1,78 @@
+package com.linecorp.linesdk.openchat.ui
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.linecorp.linesdk.openchat.OpenChatCategory
+import com.linecorp.linesdk.openchat.OpenChatParameters
+
+class OpenChatInfoViewModel : ViewModel() {
+    val chatroomName: MutableLiveData<String> = MutableLiveData()
+    val profileName: MutableLiveData<String> = MutableLiveData()
+    val description: MutableLiveData<String> = MutableLiveData()
+    val category: MutableLiveData<OpenChatCategory> = MutableLiveData()
+    val isSearchIncluded: MutableLiveData<Boolean> = MutableLiveData()
+
+    val isValid: MutableLiveData<Boolean> = MutableLiveData()
+    val isProfileValid: MutableLiveData<Boolean> = MutableLiveData()
```

Comment:
We can use the `MediatorLiveData` for `isValid` and `isProfileValid` to get rid of `updateValidity()`, `setChatroomName()`, `setDescription()`, and `setProfileName()` methods.
```kotlin
val isValid: MediatorLiveData<Boolean> = MediatorLiveData<Boolean>().apply {
    addSource(chatroomName) { chatroomNameString ->
        value = chatroomNameString.isNotEmpty()
    }
}

val isProfileValid: MediatorLiveData<Boolean> = MediatorLiveData<Boolean>().apply {
    addSource(profileName) { profileNameString ->
        value = profileNameString.isNotEmpty()
    }
}
```